src/server.rs

```diff
@@ -397,8 +397,14 @@ impl<O: Output> LsService<O> {
         self.output.success(id, ResponseData::Init(result));
 
         let root_path = params.root_path.map(PathBuf::from).expect("No root path");
+        let init_options: Option<InitializationOptions> = params.initialization_options
+                              .and_then(|options| serde_json::from_value(options)
```

Comment:
This could be just simplified to `serde_json::from_value(options).ok()` and not introduce 4 more lines for a `debug!` but I wanted to add a message in case deserialization fails for the options. Is that okay?